### DISCUSSION TITLE
Building deepstream samples on a Jetson target

### DISCUSSION THREAD

**Discussion by tirankas** [OE4T/meta-tegra#715] [General]
Hi Guys,
Would like to know whether compiling, on target, of deepstream samples on a Yocto based Jetson is possible?
Usually people compile these on a Ubuntu based SW, on a Jetson HW, and would need the same capability on a Yocto based.

When I try to build a deepstream sample, I get errors on missing packages like gstreamer-1.0,gstreamer-video-1.0, json-glib-1.0, include paths are not correct etc..

Thanks

**Comment by madisongh** [comment]
It's theoretically possible to do.  It's up to you to ensure you're including all the needed `-dev` packages.  Since the current deepstream recipe just installs the prebuilt binaries for the samples, rather than building them from source, looking at the DEPENDS in that recipe probably won't be sufficient for determining the exact set of needed packages.

After all that, you might still run into issues with the makefiles assuming the Ubuntu/Debian filesystem layout, which you'd have to fix up as needed.

**Comment by tirankas** [comment]
Thanks Matt.
Indeed, I will have to do this on a package by package basis, will do.

**Reply by dwalkes** [reply]
Another option is to build based on docker containers.  Here's an example repo you can use for this: https://github.com/BoulderAI/deepstream-nvdsanalytics

**Comment by tirankas** [comment]
Thanks, sounds interesting.
I ended up using tools-sdk and tools-debug which brought all dev packages and I was able to compile deepstream sources with this. Running vscode on the host environment, using vscode plugins on the target, it looks good.
I agree using a docker might even be better, will try this approach too.
